src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -111,6 +112,115 @@ def test_gen_dims_error(monkeypatch):
     assert "unsupported gen_dims index 'invalid'" in str(excinfo)
 
 
+def test_gen_datatype():
+    '''Check the gen_datatype function produces the expected datatypes.'''
+    for datatype in ["real", "integer", "character"]:
+        symbol = Symbol("dummy", datatype)
+        assert gen_datatype(symbol) == datatype
+    symbol = Symbol("dummy", "boolean")
+    assert gen_datatype(symbol) == "logical"
+
+
+@pytest.mark.parametrize(
+    "datatype,precision,result",
+    [("real", None, "real"),
+     ("integer", 8, "integer*8"),
+     ("real", 16, "real*16"),
+     ("real", Symbol.Precision.DOUBLE, "double precision"),
+     ("integer", Symbol("i_def", "integer"), "integer(kind=i_def)"),
+     ("real", Symbol("r_def", "integer"), "real(kind=r_def)")])
+def test_gen_datatype_precision(datatype, precision, result):
+    '''Check the gen_datatype function produces the expected datatypes when
+    precision is specified.
+
+    '''
+    symbol = Symbol("dummy", datatype, precision=precision)
+    assert gen_datatype(symbol) == result
+
+
+# Note, in isolation and when only running tests from the psyir
+# directory, this test does not fail.
+@pytest.mark.xfail(reason="issue #582 backend logging output is affected by "
```

Comment:
I don't think there's a need for a new issue - just refer to #11 here as it's something that will need to be addressed when we implement logging.

Reply:
OK. I've changed the reference and commented out the test.